### PR TITLE
sby: 2021.11.30 -> 0.44, run tests, rename from symbiyosys

### PR DESCRIPTION
**Pull request by RCoeurjoly**
## Description of changes

Create sby derivation based on SymbiYosys, with some fixes. Remove SymbiYosys.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pbsds** [comment]
please add an entry in `pkgs/top-level/aliases.nix`

File: pkgs/by-name/sb/sby/package.nix [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
 
+let
+  pythonEnv = python3.withPackages (ps: with ps; [ click ]);
+in
+
 stdenv.mkDerivation {
-  pname = "symbiyosys";
-  version = "2021.11.30";
+  pname = "sby";
+  version = "yosys-0.44";
```
Comment (pbsds) [inline pkgs/by-name/sb/sby/package.nix:13]:
```suggestion
  version = "0.44";
```

File: pkgs/by-name/sb/sby/package.nix [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
 
   src = fetchFromGitHub {
     owner = "YosysHQ";
-    repo  = "SymbiYosys";
-    rev   = "b409b1179e36d2a3fff66c85b7d4e271769a2d9e";
-    hash  = "sha256-S7of2upntiMkSdh4kf1RsrjriS31Eh8iEcVvG36isQg=";
+    repo  = "sby";
+    rev   = "yosys-0.44";
```
Comment (pbsds) [inline pkgs/by-name/sb/sby/package.nix:18]:
```suggestion
    rev   = "yosys-${version}";
```

**Review by pbsds — COMMENTED** [review]
please also configure `nix-update-script` with the correct tag prefix (`yosys-`)

**Comment by RCoeurjoly** [comment]
> please add an entry in `pkgs/top-level/aliases.nix`

Done. Thanks for the review!

**Comment by RCoeurjoly** [comment]
> please also configure `nix-update-script` with the correct tag prefix (`yosys-`)

Done.

File: pkgs/by-name/sb/sby/package.nix [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    yices
+    z3
+    aiger
+  ];
+  checkPhase = "make test";
+
+  passthru.updateScript = nix-update-script {
+    extraArgs = [
+      "--version-regex"
+      "yosys-(\d\.\d)"
```
Comment (pbsds) [inline pkgs/by-name/sb/sby/package.nix:87]:
```suggestion
      "yosys-([0-9].*)"
```

testing this took a while due to the issue fixed in https://github.com/Mic92/nix-update/pull/269

Comment (RCoeurjoly) [reply pkgs/by-name/sb/sby/package.nix:91]: Just out of curiosity, how did you test it?

**Comment by thoughtpolice** [comment]
Would you mind just squashing these commits all into a single commit? Some of them (for example changing the version field from `yosys-0.44` to `0.44`) aren't particularly meaningful on their own and can all just go into one, as I typically rebase it onto master anyway. LGTM otherwise, thanks

**Comment by RCoeurjoly** [comment]
> Would you mind just squashing these commits all into a single commit? Some of them (for example changing the version field from `yosys-0.44` to `0.44`) aren't particularly meaningful on their own and can all just go into one, as I typically rebase it onto master anyway. LGTM otherwise, thanks

Done! Thanks for the review!

**Review by thoughtpolice — APPROVED** [review]

File: pkgs/by-name/sb/sby/package.nix [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+  passthru.updateScript = nix-update-script {
+    extraArgs = [
+      "--version-regex"
+      "yosys-([0-9].*)"
+    ];
+  };
+
+  meta = {
+    description = "SymbiYosys (sby) -- Front-end for Yosys-based formal verification flows";
```
Comment (SuperSandro2000) [inline pkgs/by-name/sb/sby/package.nix:96]: The description should not repeat the package name. See contributing docs

File: pkgs/by-name/sb/sby/package.nix [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    boolector
+    yices
+    z3
+    aiger
+  ];
+  checkPhase = ''
+    runHook preCheck
+    make test
+    runHook postCheck
+  '';
```
Comment (SuperSandro2000) [inline pkgs/by-name/sb/sby/package.nix:86]: Isn't this already done by the default checkPhase when a makefile is detected?

File: pkgs/by-name/sb/sby/package.nix [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+  buildInputs = [
+    pythonEnv
+    yosys
+    boolector
+    yices
+    z3
+    aiger
+  ];
+
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/by-name/sb/sby/package.nix:41]: This breaks patches

File: pkgs/by-name/sb/sby/package.nix [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    patchShebangs .
```
Comment (SuperSandro2000) [inline pkgs/by-name/sb/sby/package.nix:42]: This should be a bit more explicit which files need to be patched

File: pkgs/by-name/sb/sby/package.nix [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+  '';
+
+  doCheck = true;
+  nativeCheckInputs = [
+    pythonEnv
+    yosys
+    boolector
+    yices
+    z3
+    aiger
```
Comment (SuperSandro2000) [inline pkgs/by-name/sb/sby/package.nix:80]: Aren't those duplicated with buildInputs?

**Comment by SuperSandro2000** [comment]
I don't think that PR should have been merged in this state. There are many things which should have been fixed.

File: pkgs/by-name/sb/sby/package.nix (hunk 4 above)
Comment (pbsds) [reply pkgs/by-name/sb/sby/package.nix:96]: Pr welcome

Comment (RCoeurjoly) [reply pkgs/by-name/sb/sby/package.nix:96]: I opened this: https://github.com/NixOS/nixpkgs/pull/336519

**Comment by thoughtpolice** [comment]
Basically all of the complaints are just as valid against the original expression since it was renamed and slightly modified here, I don't subscribe to the "Please cleanup 100 other unrelated things" philosophy most of the time, but I understand this is a chance to fix it at least. In either case we can take this to #336519

**Comment by pbsds** [comment]
> I don't think that PR should have been merged in this state. There are many things which should have been fixed.

I agree with thoughtpolice, this is essentially a bump and rename: a strict improvement with a clear scope. For that we should say thank you and merge it with as little friction as possible.
Tasking contributors to "btw you must refactor and fix all these other unrelated issues, otherwise I'll hold this PR hostage" is not very inviting.

File: pkgs/by-name/sb/sby/package.nix (hunk 3 above)
Comment (pbsds) [reply pkgs/by-name/sb/sby/package.nix]: `nix-update -u sby` :)

**Pull request merged**